### PR TITLE
Support for reading CAN timestamp

### PR DESCRIPTION
**Pull request by FabianInostroza**
This enable us to get the hardware timestamp of a CAN message.

### CODE REVIEW

File: lib/stm32/can.c [hunk 1]
```diff
@@ -522,6 +523,9 @@ void can_receive(uint32_t canport, uint8_t fifo, bool release, uint32_t *id,
 	 * memory to the fast internal RAM
 	 */
 
+	if( timestamp )
+		*timestamp = (CAN_RDTxR(canport, fifo_id) & CAN_RDTxR_TIME_MASK) >> CAN_RDTxR_TIME_SHIFT;
+
```
Comment (karlp) [inline lib/stm32/can.c]: Please follow the existing style here.  try "make stylecheck" on your new branch with this code.  The code is good, but just needs this style cleanup.  Please expand the documentation the timestamp parameter too, that it's optional and to provide NULL.  (the unsigned int16 portion can be dropped from the doxygen, it's inferred from the types automatically)

(You need {} even for single line ifs, and the spacing is wrong)

**Comment by karlp** [comment]
please add "stm32: can: " as a prefix on these commit subjects to.  YOu can just push -f to this branch, and github will upate the PR automatically.

**Comment by FabianInostroza** [comment]
Done.

**Comment by karlp** [comment]
Thanks, included in my staging now.

**Comment by karlp** [comment]
This has finally landed witth my staging 2017.4 branch

**Pull request closed without merging**